tests/daemon/unittest_dbus_model.cc

```diff
@@ -0,0 +1,294 @@
+/**
+ * @file        unittest_dbus_model.cc
+ * @date        29 Jul 2022
+ * @brief       Unit test for DBus Model interface
+ * @see         https://github.com/nnstreamer/api
+ * @author      Sangjung Woo <sangjung.woo@samsung.com>
+ * @bug         No known bugs
+ */
+
+#include <gio/gio.h>
+#include <gtest/gtest.h>
+#include <errno.h>
+
+#include "model-dbus.h"
+#include "../../daemon/includes/dbus-interface.h"
+
+/**
+ * @brief Test base class of DBus Model interface
+ */
+class DbusModelTest : public::testing::Test
+{
+protected:
+  GTestDBus *dbus;
+  GDBusProxy *server_proxy;
+
+public:
+  /**
+   * @brief Setup method for each test case.
+   */
+  void SetUp() override
+  {
+    gchar *services_dir = g_build_filename (g_get_current_dir (), "tests/services", NULL);
+    dbus = g_test_dbus_new (G_TEST_DBUS_NONE);
+    g_test_dbus_add_service_dir (dbus, services_dir);
+    g_test_dbus_up (dbus);
+
+    GError *error = NULL;
+    server_proxy = g_dbus_proxy_new_for_bus_sync (
+      G_BUS_TYPE_SESSION,
+      G_DBUS_PROXY_FLAGS_NONE,
+      NULL,
+      DBUS_ML_BUS_NAME,
+      DBUS_MODEL_PATH,
+      DBUS_MODEL_INTERFACE,
+      NULL,
+      &error);
+
+    if (!server_proxy || error) {
+      if (error) {
+        g_critical ("Error Message : %s", error->message);
+        g_clear_error (&error);
+      }
+    }
+
+    g_free (services_dir);
+  }
+
+  /**
+   * @brief Teardown method for each test case.
+   */
+  void TearDown() override
+  {
+    if (server_proxy)
+      g_object_unref (server_proxy);
+
+    g_test_dbus_down (dbus);
+    g_object_unref (dbus);
+  }
+};
+
+/**
+ * @brief Call the 'set_path' DBus method to store the model name and its file path.
+ */
+TEST_F (DbusModelTest, set_path_00_p)
+{
+  int ret;
+  const gchar *name = "mobilenetv3";
+  const gchar *path = "/opt/usr/shared/mobilenet_v3.pb";
+  MachinelearningServiceModel *proxy = nullptr;
+  GError *error = nullptr;
+
+  proxy = machinelearning_service_model_proxy_new_for_bus_sync (
+    G_BUS_TYPE_SESSION,
+    G_DBUS_PROXY_FLAGS_NONE,
+    DBUS_ML_BUS_NAME,
+    DBUS_MODEL_PATH,
+    NULL, &error);
+  if (error != nullptr) {
+    g_critical ("Error : %s", error->message);
+    g_error_free (error);
+    FAIL();
+  }
```

Comment:
These lines are repeated. Is there any reason why they are not included in the test fixture?

Reply:
My fault. I will move this repeated code to `SetUp()` method.